GameWorld.Gen/Perlin3D.cs

```diff
@@ -0,0 +1,169 @@
+            int floorZ = (int)Math.Floor(z);
+
+            // get relative XYZ coordinates of point in cell
+            double relX = x - floorX;
+            double relY = y - floorY;
+            double relZ = z - floorZ;
+
+            //gradients of cube vertices
+            VectorD g000 = getGradient(floorX, floorY, floorZ);
+            VectorD g001 = getGradient(floorX, floorY, floorZ + 1);
+            VectorD g010 = getGradient(floorX, floorY + 1, floorZ);
+            VectorD g011 = getGradient(floorX, floorY + 1, floorZ + 1);
+            VectorD g100 = getGradient(floorX + 1, floorY, floorZ);
+            VectorD g101 = getGradient(floorX + 1, floorY, floorZ + 1);
+            VectorD g110 = getGradient(floorX + 1, floorY + 1, floorZ);
+            VectorD g111 = getGradient(floorX + 1, floorY + 1, floorZ + 1);
+
+            // noise contribution from each of the eight corner
+            double n000 = g000 * new VectorD(new[] { relX, relY, relZ });
+            double n100 = g100 * new VectorD(new[] { relX - 1, relY, relZ });
+            double n010 = g010 * new VectorD(new[] { relX, relY - 1, relZ });
+            double n110 = g110 * new VectorD(new[] { relX - 1, relY - 1, relZ });
+            double n001 = g001 * new VectorD(new[] { relX, relY, relZ - 1 });
+            double n101 = g101 * new VectorD(new[] { relX - 1, relY, relZ - 1 });
+            double n011 = g011 * new VectorD(new[] { relX, relY - 1, relZ - 1 });
+            double n111 = g111 * new VectorD(new[] { relX - 1, relY - 1, relZ - 1 });
+
+            // compute the fade curve value for each x, y, z
+            double u = BlendingCurve(relX);
+            double v = BlendingCurve(relY);
+            double w = BlendingCurve(relZ);
+
+            // interpolate along x the contribution from each of the corners
+            double nx00 = interpolation(n000, n100, u);
+            double nx01 = interpolation(n001, n101, u);
+            double nx10 = interpolation(n010, n110, u);
+            double nx11 = interpolation(n011, n111, u);
+
+            // interpolate the four results along y
+            double nxy0 = interpolation(nx00, nx10, v);
+            double nxy1 = interpolation(nx01, nx11, v);
+
+            // interpolate the two last results along z
+            double nxyz = interpolation(nxy0, nxy1, w);
+
+            return nxyz;
+        }
+
+        public double getMultioctave3DNoiseValue(double x, double y, double z, uint startOctaveNumber, uint octaveCount)
+        {
+            double res = 0;
+            for (uint i = startOctaveNumber; i < (startOctaveNumber + octaveCount); ++i)
+            {
+                var powOf2 = FastPow(2, i);
+
+                res += (powOf2 * get3DNoiseValue(x / powOf2, y / powOf2, z / powOf2));
+            }
+            return res;
+        }
+
+        public double getMultioctave3DNoiseValueFromSphere(int x, int y, int z, uint startOctaveNumber, uint octaveCount, uint radius)
```

Comment:
This function is not needed, you can just pass tile center coords to getMultioctave3DNoiseValue and get the proper value for sphere.